apollo-federation/src/connectors/models.rs

```diff
@@ -214,6 +216,20 @@ impl Connector {
             .is_success
             .as_ref()
             .or_else(|| source.and_then(|s| s.is_success.as_ref()));
+        let fragments = source.map(|s| s.fragments.clone());
+
+        let mut selection = connect.selection;
+        if let (Some(fragments), Some(source_name)) = (fragments, source_name.as_ref()) {
+            for (name, frag_selection) in fragments {
+                let fragment_query = format!("{FRAGMENT_INLINE}.{source_name}.{name}");
```

Comment:
For discussion: do we really need the `$fragments.source.Fragment` structure, or could we just flatten the fragments into a single `$fragments.Fragment` namespace (and error if there are cross-`@source` collisions)?